rir/src/ir/Compiler.cpp

```diff
@@ -475,6 +478,11 @@ bool compileSpecialCall(Context& ctx, SEXP ast, SEXP fun, SEXP args_) {
             //            if (getter == symbol::Bracket)
         }
 
+        if (superAssign) {
```

Comment:
can you move the guard after the condition (only inserting it if needed) instead of removing it afterwards?